packages/next-swc/crates/core/src/relay.rs

```diff
@@ -0,0 +1,169 @@
+use serde::Deserialize;
+use swc_atoms::{js_word, JsWord};
+use swc_common::{FileName, Span};
+use swc_ecmascript::ast::*;
+use swc_ecmascript::visit::{Fold, FoldWith};
+
+#[derive(Clone, Debug, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub enum RelayLanguageConfig {
+    Typescript,
+    Flow,
+}
+
+struct Relay {
+    file_name: FileName,
+    relay_config_for_tests: Option<ProjectConfig>,
+}
+
+fn pull_first_operation_name_from_tpl(tpl: &TaggedTpl) -> Option<&str> {
+    tpl.tpl
+        .quasis
+        .iter()
+        .filter_map(|quasis| {
+            let split_content = quasis.raw.value.split(" ").collect::<Vec<&str>>();
+
+            let operation = split_content
+                .chunks(2)
+                .filter_map(|slice| {
+                    if slice.len() == 1 {
+                        return None;
+                    }
+
+                    let word = slice[0];
+                    let next_word = slice[1];
+
+                    if word == "query"
+                        || word == "subscription"
+                        || word == "mutation"
+                        || word == "fragment"
+                    {
+                        return Some(next_word);
+                    }
+
+                    None
+                })
+                .next();
+
+            return operation;
+        })
+        .next()
+}
+
+fn build_require_expr_from_path(path: String) -> Expr {
+    Expr::Call(CallExpr {
+        span: Default::default(),
+        callee: Callee::Expr(Box::new(Expr::Ident(Ident::new(
+            js_word!("require"),
+            Span::default(),
+        )))),
+        args: vec![ExprOrSpread {
+            spread: None,
+            expr: Box::new(Expr::Lit(Lit::Str(Str {
+                span: Default::default(),
+                value: JsWord::from(path),
+                has_escape: false,
+                kind: Default::default(),
+            }))),
+        }],
+        type_args: None,
+    })
+}
+
+impl Fold for Relay {
+    fn fold_expr(&mut self, expr: Expr) -> Expr {
+        let expr = expr.fold_children_with(self);
+
+        match &expr {
+            Expr::TaggedTpl(tpl) => {
+                if let Some(built_expr) = self.build_call_expr_from_tpl(tpl) {
+                    built_expr
+                } else {
+                    expr
+                }
+            }
+            _ => expr,
+        }
+    }
+}
+
+// This is copied from https://github.com/facebook/relay/blob/main/compiler/crates/relay-compiler/src/build_project/generate_artifacts.rs#L251
+// until the Relay team exposes it for external use.
+fn path_for_artifact(project_config: &ProjectConfig, definition_name: &str) -> PathBuf {
+    let source_file_location_key = SourceLocationKey::Standalone {
+        // TODO: Figure out why passing in the actual path here causes the requires to be an
+        // absolute path.
+        path: StringKey::from_str("NA").unwrap(),
```

Comment:
@alunyov Can you chime in here? If I pass a proper path here, the resulting require uses an absolute path from the root of my fs.